docs/developer/develop-minikube.md

```diff
@@ -0,0 +1,178 @@
+
+Open up the turbinia folder in VSCode.
+
+### Prepare Cluster
+Open a terminal (inside VSCode is the easiest, but any terminal will do) and let's configure skaffold, the local cluster and the additional helm repository for Redis.
+
+    $ skaffold config set --global local-cluster true
+    $ eval $(minikube -p minikube docker-env)
+    $ helm repo add bitnami https://charts.bitnami.com/bitnami
+
+### Verify Setup
+Execute a build with skaffold (from the root of the cloned Turbinia Github repository)
+
+    $ skaffold build
+
+This will build a Turbinia Server container image succesfully if skaffold has been correctlty setup and configured as described above.
+
+![Skaffold Build](../images/cloudcode-skaffoldbuild.png)
+
+### Install the Turbinia Client
+We will install the Turbinia client into a Python virtual environment to be able to control Turbinia during our development workflow.
+
+    $ python -m venv .venv (or use your favorite virtual env manager)
+    $ ./venv/bin/activate
+    $ pip install turbinia-client
+
+Create the Turbinia Client configuration file in `$HOME/.turbinia_api_config.json` using the base configuration from [here](
+https://pypi.org/project/turbinia-client/).
+
+![Turbinia Client](../images/cloudcode-turbiniaclient.png)
+
+### Run
+Now we are ready to run the development cluster of Turbinia. This will startup the Turbinia API server, the worker and the server in the local minikube k8s cluster.
+
+    $ skaffold dev
+
+NOTE: if one of the services fails to deploy, try again. Sometimes a time-out occurs due to Redis starting too slow.
+
+The Turbinia WebUI should now be available on http://localhost:8000.
+
+![Turbinia WebUI](../images/cloudcode-webui.png)
+
+
+### Verify debugging and hot-reloading
+#### Debugging
+The Turbinia containers are started with the python debugger enabled and listening on port 10000 (worker) and 20000 (server). VSCode launch debug configurations have been provided.
+
+While the cluster is running in development mode with skaffold you can use those launch configuration to connect to the worker and server (yes, at the same time if you want that!), set breakpoints and inspect watches.
+
+![Turbinia VSCode Launch Configurations](../images/cloudcode-vscodedebug.png)
+
+#### Hot-reloading
+Hot reloading has been enabled through the python `jurigged` [module](https://github.com/breuleux/jurigged) and will monitor for any change on the `/home/turbinia` folder in the containers. When you change a file in VSCode, skaffold will sync that file into the container and `jurigged` will hot reload that file into the running process.
+
+![Turbinia Hot Reloading](../images/cloudcode-hotreload.png)
+
+NOTE: As python hot reloading of code into an already running process is tricky it may not work in all cases (see Caveats section [here](https://github.com/breuleux/jurigged?tab=readme-ov-file)). If that happens let skaffold rebuild the container by touching the `skaffold.yaml`file.
```

Comment:
I've noticed this as well, it's hard to tell what can even be hot reloaded where I feel like I'd just need to do this each time. Is there any benefit to hot reloading with our code base? I haven't tested it with analyzers and what not just the server so may have been better success there?

Reply:
Hot reloading works across the complete code base. Something to keep in mind:
- hot reloading is loading new python byte code for a file in memory
- hot reloading will not re-execute anything
- so you need to make sure you do something with Turbinia (create new request, download results or any other action) to hit the code path you changed and got hot reloaded

Reply:
Added an explanation to the text.